src/App.tsx

```diff
@@ -1,16 +1,26 @@
 import './App.scss';
+import React, { useState } from 'react';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
+import { Movie } from './types/Movie';
 import moviesFromServer from './api/movies.json';
 
-export const App = () => {
+export const App: React.FC = () => {
+  const [moviesList, setMoviesList] = useState<Movie[]>(moviesFromServer);
+
+  const onAddMovie = (movie: Movie) => {
+    setMoviesList([...moviesList, movie]);
+  };
+
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={moviesList} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie
+          onAdd={onAddMovie}
```

Comment:
```suggestion
          onAdd={onAddMovie}
          key={count}
```
Add and change key counter here. It will re-render component from scratch with default values and no errors